src/org/hammerhead226/sharkmacro/motionprofiles/Profile.java

```diff
@@ -112,10 +107,8 @@ public Profile(String[][] leftProfile, String[][] rightProfile, TalonSRX leftTal
 	 *            the PID gains profile to use to execute the right motion profile
 	 */
 	public void execute(int leftGainsProfile, int rightGainsProfile) {
-		left = new ProfileHandler(leftProfile, leftTalon, leftGainsProfile);
-		right = new ProfileHandler(rightProfile, rightTalon, rightGainsProfile);
-		left.execute();
-		right.execute();
+		handler = new ProfileHandler(leftProfile, rightProfile, leftTalon, rightTalon, leftGainsProfile, rightGainsProfile);
```

Comment:
You could potentially do an array of profiles, talons, and gains profiles here. Not a stopper.